pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -210,6 +221,11 @@ func (adc *attachDetachController) Run(stopCh <-chan struct{}) {
 	defer runtime.HandleCrash()
 	glog.Infof("Starting Attach Detach Controller")
 
+	if !kcache.WaitForCacheSync(stopCh, adc.podsSynced, adc.nodesSynced, adc.pvcsSynced, adc.pvsSynced) {
```

Comment:
I was going through and adding them throughout. I will revisit all my changes and I guess only wait if the code in master is already waiting?